src/main/Scheduler.java

```diff
@@ -69,29 +71,31 @@ public void run() {
 					
 					for(DestinationEvent destEvent : destinationEvents) {
 						if(destEvent.getDestination() == arrivalEvent.getCurrentFloor()) {
-							keepGoing = !destinationEvents.isEmpty();
-							//set the floor event to null since thats what elevator wants. 
-							currentFloorEvent = null; 
 							destinationEventFlag = true; 
 							break; 
 						}
 					}
+					
+					//checks if the elevator should keep going. 
+					elevatorKeepsGoing = !destinationEvents.isEmpty();
 										
 					if(!floorEventFlag || !destinationEventFlag) {
 						//No-stop
-						schedulerEvent = new SchedulerEvent(arrivalEvent.getDirection(), LocalTime.now());
+						schedulerEvent = new SchedulerEvent(currentFloorEvent.getDirection(), LocalTime.now());
 					}
 					else if(destinationEventFlag && floorEventFlag) {
-						schedulerEvent = new SchedulerEvent(arrivalEvent.getCurrentFloor(), keepGoing, true, 
-								true, currentFloorEvent, arrivalEvent.getDirection(), LocalTime.now());
+						schedulerEvent = new SchedulerEvent(arrivalEvent.getCurrentFloor(), elevatorKeepsGoing, true, 
+								true, currentFloorEvent, currentFloorEvent.getDirection(), LocalTime.now());
 					}
 					else if(destinationEventFlag) {
-						schedulerEvent = new SchedulerEvent(arrivalEvent.getCurrentFloor(), keepGoing, true, 
-								false, currentFloorEvent, arrivalEvent.getDirection(), LocalTime.now());
+						//set the floor event to null since thats what elevator wants. 
+						currentFloorEvent = null; 
+						schedulerEvent = new SchedulerEvent(arrivalEvent.getCurrentFloor(), elevatorKeepsGoing, true, 

```

Comment:
technically you could take out the comment and line 92, if currentFloor is never set by the FloorEvent for loop, it'll be null. If floorevent isn't null here then there's an issue

Reply:
fixed